scripts/superdesk/directives/sdPasswordStrength.js

```diff
@@ -0,0 +1,111 @@
+(function() {
+    'use strict';
+
+    // config holds the configuration for the password strength calculator.
+    var config = {
+        MIN_LENGTH: 8,
+        ONE_LOWER: /^(?=.*[a-z])/g,
+        ONE_UPPER: /^(?=.*[A-Z])/g,
+        ONE_NUMBER: /^(?=.*[0-9])/g,
+        ONE_OTHER: /^(?=.*[^0-9^a-z^A-Z])/g,
+        MIN_STRENGTH: 3
+    };
+
+    // styles holds each of the strength labels by index along with the class
+    // to be added to the indicator.
+    var styles = [
+        {txt: 'Short', cls: 'red'},
+        {txt: 'Weak', cls: 'red'},
+        {txt: 'Better', cls: 'yellow'},
+        {txt: 'OK', cls: 'green'},
+        {txt: 'Strong', cls: 'green'}
+    ];
+
+    // helpText holds the text that will be shown when the user hovers over the
+    // informational icon.
+    var helpText = 'Must be ' + config.MIN_LENGTH + ' characters long and ' +
```

Comment:
would be good to translate this, ideally with `$interpolate` like [here](https://github.com/superdesk/superdesk-client-core/blob/master/scripts/superdesk-authoring/authoring.js#L2477)